db/collection.go

```diff
@@ -451,18 +459,18 @@ func (t *Txn) Discard() {
 	t.discarded = true
 }
 
-func getInstanceID(t []byte) core.InstanceID {
+func getInstanceID(t []byte) (core.InstanceID, error) {
 	partial := &struct{ ID *string }{}
 	if err := json.Unmarshal(t, partial); err != nil {
-		log.Fatalf("error when unmarshaling json instance: %v", err)
+		return core.EmptyInstanceID, err
```

Comment:
`log.Fatal` seems strange to me. I'm not sure why we'd ever want to actually just kill the process when there is an error we can return to the caller. I made this change here since I was working around/in this code, but this `log.Fatal` is used other places as well. If there is a good reason to revert this, let me know.

Reply:
There are cases where fatal makes sense. For example, "I've encountered an error that indicates a fatal flaw on logic and I should not be used anymore". There are a number of places where fatal is used like that in here. This one around getting the instance ID _could_ be like that. The system allowed the ID to enter, if it can't get it back out, something bad happened that can't be recovered from. It's nice to have a distinction in the code the separates these kinds of things. But, I'm not super partial to one direction for this one.

Reply:
It would kill their daemon, hopefully they would report the fatal bug.